### ISSUE TITLE
Snowflake, a censorship circumvention system using temporary WebRTC proxies (USENIX Security 2024)

### ISSUE THREAD

**Issue by wkrp** [net4people/bbs#366]
Snowflake, a censorship circumvention system using temporary WebRTC proxies
Cecylia Bocovich, Arlo Breault, David Fifield, Serene, Xiaokang Wang
https://censorbib.nymity.ch/#Bocovich2024a
[Online HTML](https://www.bamsoftware.com/papers/snowflake/)
[usenix.org page](https://www.usenix.org/conference/usenixsecurity24/presentation/bocovich) (will have slides and presentation video eventually)
[PDF](https://github.com/net4people/bbs/files/15385407/snowflake.pdf)

This paper describes the [Snowflake](https://snowflake.torproject.org/) circumvention system: how it works, and a retrospective of operation since 2021, including reactions by censors. The principle behind Snowflake's blocking resistance is the use of numerous (currently on the order of 100,000) non-permanent proxies that communicate with censored clients using [WebRTC](https://en.wikipedia.org/wiki/WebRTC) protocols. Snowflake proxies are simple and cheap to operate in comparison to a traditional proxy server; they can even run in a web page or browser extension; there is also a command-line version and a Snowflake proxy built into Orbot. The large and dynamic pool of proxy IP addresses is supposed to be difficult for a censor to enumerate and block, and the use of common protocols (WebRTC is often used in web browsers for audio and video communication) makes it more expensive to block by protocol.

The Snowflake system involves a many interacting components. A censored client contacts a central server called the _broker_, which matches the client with one of the currently available proxies and helps exchange the information that the client and proxy need in order to establish a connection ([WebRTC signaling](https://www.rfc-editor.org/rfc/rfc8825.html#section-2.2)). The censor is assumed to block the client from contacting the broker directly, so the client must communicate with the broker using a blocking-resistant rendezvous protocol. (The currently implemented rendezvous methods are domain fronting, [AMP cache](https://gitlab.torproject.org/tpo/anti-censorship/pluggable-transports/snowflake/-/merge_requests/50), and [Simple Queue Service](https://gitlab.torproject.org/tpo/anti-censorship/pluggable-transports/snowflake/-/merge_requests/214).) The client starts sending information to the proxy, but the proxy doesn't exit the client's traffic directly: instead, it forwards the traffic to a permanent, centralized _bridge_, which is what actually delivers the traffic to the client's desired destination. The traffic stream is encrypted between the client and the bridge, so proxies cannot eavesdrop on or tamper with user traffic. Having a separate bridge also ensures that client traffic is not wrongly attributed to volunteer proxies. The protocol stack features a [Turbo Tunnel](https://github.com/net4people/bbs/issues/9) layer (underneath the WebRTC) so that clients can switch from one proxy to another without interruption. (Proxies are allowed to go offline at any time, even when they are being used by a client.)

[Section 4](https://www.bamsoftware.com/papers/snowflake/#experience) is a blow-by-blow account of how Snowflake has fared in deployment, and how the number of users and proxies have been affected by world events. The authors provide detailed case studies of reactions of censors in four countries. At the end of 2021, users in Russia became the largest contingent of Snowflake users after an incident of [blocking of Tor servers and protocols](https://github.com/net4people/bbs/issues/97). In 2022, a large number of people began using Snowflake in Iran after [shutdowns and blocking there](https://github.com/net4people/bbs/issues/125). Snowflake was [briefly blocked by TLS fingerprinting](https://github.com/net4people/bbs/issues/131) but recovered, and the majority of Snowflake users are in Iran to this day. In China there has not been much evidence of attempts to block Snowflake, with only [one](https://github.com/net4people/bbs/issues/249) or [two](https://github.com/net4people/bbs/issues/255) minor incidents recorded. In Turkmenistan, Snowflake has not been very successful, only ever having a maximum of a few dozen simultaneous users; nevertheless there is a history of blocking and unblocking events.

**Comment by wkrp** [comment]
This is the finished version of the paper. Work-in-progress drafts and discussion of them are in #296.

**Comment by Tw-C** [comment]
@wkrp I came across this state funded research after suspecting if Snowflake has been a lost cause, and guess what I found!
https://www.mdpi.com/2076-3417/13/1/622

**Comment by wkrp** [comment]
> @wkrp I came across this state funded research after suspecting if Snowflake has been a lost cause, and guess what I found! https://www.mdpi.com/2076-3417/13/1/622

Yes! We in fact cited this research in the [Fingerprinting section of the paper](https://www.bamsoftware.com/papers/snowflake/#p57) (Chen et al.).

**Comment by wkrp** [comment]
The Snowflake conference presentation video (14 minutes) is now online. This is the original from YouTube:

https://www.youtube.com/watch?v=rZYfrj2iqYE&list=PLbRoZ5Rrl5ldQ2K_dpmPKHEyRgyf5JSxd&index=149

This one has English captions and a full transcript:

https://www.bamsoftware.com/talks/snowflake-usenix2024/#video
